src/App.js

```diff
@@ -2,18 +2,53 @@ import React from 'react';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
+
 function App(){
+
   const [todoList, setTodoList] = React.useState([]);
+  const [isLoading, setIsLoading] = React.useState(true);
+
+  React.useEffect (() => {
+    new Promise((resolve,reject) => 
+    setTimeout(
+      () => resolve ({data: {todoList: JSON.parse(localStorage.getItem('savedTodoList')) } }),
+      2000
+    )
+    ).then(result => {
+      setTodoList(result.data.todoList);
+      setIsLoading(false);
+      });
+  }, []);
+
+  React.useEffect (() => {
+    if (!isLoading) {
+      localStorage.setItem('savedTodoList', JSON.stringify(todoList))
+    }
+  }, [todoList] );
```

Comment:
You need to add isLoading as dependency to this useEffect, so the result would be `[todoList,isLoading]`

Reply:
Thank you so much. No worries, I too have been having my fair share of challenges. I will make the necessary changes and merge.